### ISSUE TITLE
Provide request signature calculated by usage plugin in context

### ISSUE THREAD

**Issue by theJC** [apollographql/router#4558]
**Is your feature request related to a problem? Please describe.**
The instrumentation required by our company leverages the exact same operation signature as generated by the UsageReporting plugin and exposed on the span as apollo_private.operation_signature via [plugins/telemetry/mod.rs ](https://github.com/apollographql/router/blob/2c8115d7b67253686871473c4c4fe93b53476ad1/apollo-router/src/plugins/telemetry/mod.rs#L435-L443)

**Describe the solution you'd like**
Please provide the ability to expose and make available the string representation of the operation via the context

**Describe alternatives you've considered**
We have considered repeating the operation signature calculation (and caching) a second time in our own plugin code.... this will add additional processing and memory overhead that we do not want to incur in Router.

**Additional context**
Currently these fields are available in the context:
`operation_name`
`operation_kind`

so something like `operation_signature` seems like a candidate key to make available via the context.

**Comment by o0Ignition0o** [comment]
Hi, thanks for opening this issue!

We introduced the context's `apollo_operation_id` in #3586.

It behaves the same as [operation signature](https://www.apollographql.com/docs/graphos/metrics/operation-signatures/#logging-signatures-at-runtime), and is used in studio as well:

```
https://studio.apollographql.com/graph/<your_graph_variant>/variant/<your_graph_variant>/operations?query=<apollo_operation_id>
```

Would this fit your usecase?

**Comment by theJC** [comment]
Unfortunately not, it appears that the operation id is just a hash of the signature, where we require the actual signature itself for our own observability purposes.

usage_reporting.stats_report_key

**Comment by lrlna** [comment]
We are making some changes to the operation signature soon, and we'd want to wait for those to land before we expose this API.

**Comment by theJC** [comment]
Q: Double checking for planning purposes..... will the resulting signature for the same query in Apollo Gateway and Apollo Router still stay the same, its just more about the implementation details thats changing?

**Comment by bonnici** [comment]
It will be the same signature and "Apollo operation ID".

**Comment by Geal** [comment]
after discussion, it appears this won't be blocked by the operation signature changes, so I'm opening a PR to make it available in #4864
